cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -187,7 +190,19 @@ protected BaseHiveExploreService(TransactionSystemClient txClient, DatasetFramew
   }
 
   protected HiveConf getHiveConf() {
-    return new HiveConf();
+    HiveConf conf = new HiveConf();
+    // Read delegation token if security is enabled.
+    if (ShimLoader.getHadoopShims().isSecurityEnabled()) {
+      conf.set(HIVE_METASTORE_TOKEN_KEY, HiveAuthFactory.HS2_CLIENT_TOKEN);
+
+      // This is already done in Hive's ExecDriver, but we need to do it again. This will not work
+      // when secure impersonation is used.
```

Comment:
`mapreduce.job.credentials.binary` is added by Hive only if Kerberos credentials are present and impersonation is enabled. However, in our case we don't have Kerberos credentials for Explore service. Hence it will not be automatically added by Hive, instead we have to add it ourselves. 

Also add the following TODO -
TODO: When Explore does secure impersonation this has to be the tokens of the user, and not the tokens of the service itself.